rosidl_typesupport_introspection_c/cmake/rosidl_typesupport_introspection_c_generate_interfaces.cmake

```diff
@@ -14,86 +14,61 @@
+foreach(_idl_tuple ${rosidl_generate_interfaces_IDL_TUPLES})
+  string(REGEX REPLACE ":([^:]*)$" "/\\1" _abs_idl_file "${_idl_tuple}")
+  get_filename_component(_parent_folder "${_abs_idl_file}" DIRECTORY)
   get_filename_component(_parent_folder "${_parent_folder}" NAME)
-  get_filename_component(_msg_name "${_idl_file}" NAME_WE)
-  string_camel_case_to_lower_case_underscore("${_msg_name}" _header_name)
-  if(_extension STREQUAL ".msg")
-    set(_allowed_parent_folders "msg" "srv" "action")
-    if(NOT _parent_folder IN_LIST _allowed_parent_folders)
-      message(FATAL_ERROR "Interface file with unknown parent folder: ${_idl_file}")
-    endif()
-  elseif(_extension STREQUAL ".srv")
-    set(_allowed_parent_folders "srv" "action")
-    if(NOT _parent_folder IN_LIST _allowed_parent_folders)
-      message(FATAL_ERROR "Interface file with unknown parent folder: ${_idl_file}")
-    endif()
-  else()
-    message(FATAL_ERROR "Interface file with unknown extension: ${_idl_file}")
-  endif()
-  set(_generated_source_files "_generated_${_parent_folder}_source_files")
-  set(_generated_header_files "_generated_${_parent_folder}_header_files")
-  list(APPEND ${_generated_header_files}
+  get_filename_component(_idl_name "${_abs_idl_file}" NAME_WE)
+  string_camel_case_to_lower_case_underscore("${_idl_name}" _header_name)
+  list(APPEND _generated_header_files
     "${_output_path}/${_parent_folder}/${_header_name}__rosidl_typesupport_introspection_c.h")
-  list(APPEND ${_generated_source_files}
+  list(APPEND _generated_source_files
     "${_output_path}/${_parent_folder}/${_header_name}__type_support.c")
 endforeach()
 
 set(_dependency_files "")
 set(_dependencies "")
 foreach(_pkg_name ${rosidl_generate_interfaces_DEPENDENCY_PACKAGE_NAMES})
-  foreach(_idl_file ${${_pkg_name}_INTERFACE_FILES})
-  get_filename_component(_extension "${_idl_file}" EXT)
-  if(_extension STREQUAL ".msg")
+  foreach(_idl_file ${${_pkg_name}_IDL_FILES})
     set(_abs_idl_file "${${_pkg_name}_DIR}/../${_idl_file}")
     normalize_path(_abs_idl_file "${_abs_idl_file}")
     list(APPEND _dependency_files "${_abs_idl_file}")
     list(APPEND _dependencies "${_pkg_name}:${_abs_idl_file}")
-  endif()
   endforeach()
 endforeach()
 
 set(target_dependencies
   "${rosidl_typesupport_introspection_c_BIN}"
   ${rosidl_typesupport_introspection_c_GENERATOR_FILES}
+  "${rosidl_typesupport_introspection_c_TEMPLATE_DIR}/idl__rosidl_typesupport_introspection_c.h.em"
+  "${rosidl_typesupport_introspection_c_TEMPLATE_DIR}/idl__type_support.c.em"
   "${rosidl_typesupport_introspection_c_TEMPLATE_DIR}/msg__rosidl_typesupport_introspection_c.h.em"
   "${rosidl_typesupport_introspection_c_TEMPLATE_DIR}/msg__type_support.c.em"
   "${rosidl_typesupport_introspection_c_TEMPLATE_DIR}/srv__rosidl_typesupport_introspection_c.h.em"
   "${rosidl_typesupport_introspection_c_TEMPLATE_DIR}/srv__type_support.c.em"
-  ${rosidl_generate_interfaces_IDL_FILES}
+  # ${rosidl_generate_interfaces_IDL_FILES}
```

Comment:
@dirk-thomas remove or missing TODO?

Reply:
Removed in 355af2eb69ce537aa2b2b02af8b97807c6a7291e.